instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/CamelPropagationUtil.java

```diff
@@ -58,8 +58,12 @@ static void injectParent(Context context, final Map<String, Object> exchangeHead
     }
 
     @Override
-    public String get(Map<String, Object> map, String s) {
-      return (map.containsKey(s) ? map.get(s).toString() : null);
+    public String get(Map<String, Object> map, String key) {
+      return (hasValidValue(map, key) ? map.get(key).toString() : null);
+    }
+
+    private boolean hasValidValue(Map<String, Object> map, String key) {
+      return (map.containsKey(key) && (map.get(key) != null));
```

Comment:
I think that calling `containsKey()` prior to `get()` is clunky.  This `get()` is also repeated in the caller just a few lines above.  I suggest combining these methods into a simple 2-liner:
```
Object value = map.get(key);
return value == null : null : value.toString();
```

Reply:
Makes sense :)